### PR TITLE
Enable zCICS to run on Linux/MacOS

### PR DESCRIPTION
**Pull request by jyganci**
Contains files needed for zCICS to run on Linux/MacOS and continue to run on Windows.
1. Build zCICS. Windows: run cics\bat\DFHALL.BAT.  Linux/MacOS: run cics/bash/dfhall.
2. Build VSAM files used by zCICS. Windows: run cics\bat\DFHALLV.BAT. Linux/MacOS: run cics/bash/dfhallv.
3. Start zCICS. Windows: run cics\bat\Z390CICG.BAT. Linux/MacOS: run cics/bash/z390cicg.
4. When the 2 zCICS windows appear, run assembler "Hello World" transaction.
4.1. Get focus in one of the two windows.
4.2. Ctrl+C to clear the screen.
4.3. Enter GUI7 in the command text box; press Enter.
4.4. Ctrl+C to clear the screen.
5. Run the COBOL "Hello World" transaction. Similar to 4 but enter GUIC in the command text box.
6. Run other transactions, such as BED1, BEC1, VSM1, VSC1.
7. Run a CEMT transaction: Enter CEMT I SYS then press Enter.
8. Shut down zCICS: Enter CEMT P SHU

### CODE REVIEW

File: cics/DFHPCT.MLC [hunk 1]
```diff
@@ -1,21 +1,7 @@
...
-* (at your option) any later version.
-* z390 is distributed in the hope that it will be useful,
-* but WITHOUT ANY WARRANTY; without even the implied warranty of
-* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-* GNU General Public License for more details.
-*
-* You should have received a copy of the GNU General Public License 
-* along with this program; if not, see https://www.gnu.org/licenses.
-***********************************************************************
+* Copyright 2006 Automated Software Tools Corporation                 *
```
Comment (abekornelis) [inline cics/DFHPCT.MLC]: Regression of copyright notice. Please replace with the new version.

File: cics/TESTNT01.MLC [hunk 2]
```diff
@@ -1,21 +1,7 @@
...
-* but WITHOUT ANY WARRANTY; without even the implied warranty of
-* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-* GNU General Public License for more details.
-*
-* You should have received a copy of the GNU General Public License 
-* along with this program; if not, see https://www.gnu.org/licenses.
-***********************************************************************
+* Copyright 2006 Automated Software Tools Corporation                 *
+* This source code is part of z390 assembler/emulator package         *
+* The z390 package is distributed under GNU general public license    *
```
Comment (abekornelis) [inline cics/TESTNT01.MLC]: Copyright notice has regressed to prior version.

File: cics/TESTNT02.MLC [hunk 3]
```diff
@@ -1,21 +1,7 @@
...
-* but WITHOUT ANY WARRANTY; without even the implied warranty of
-* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-* GNU General Public License for more details.
-*
-* You should have received a copy of the GNU General Public License 
-* along with this program; if not, see https://www.gnu.org/licenses.
-***********************************************************************
+* Copyright 2006 Automated Software Tools Corporation                 *
+* This source code is part of z390 assembler/emulator package         *
+* The z390 package is distributed under GNU general public license    *
```
Comment (abekornelis) [inline cics/TESTNT02.MLC]: Copyright regression.

**Review by dsh33782 — DISMISSED** [review]
1.  I only saw one old copyright in TESTGUIC.CBL.
2.  To speed cics DFHALL compiles up, I think we could remove the calll to UNREF which I believe just generates report on unreferenced variables.  It could be made an option for debugging purposes. 
3. It's ok with me for now so we can get it up and tested.

**Comment by jyganci** [comment]
All,

Even though Don has approved the pull request, I will hold off doing the
merge for a few days just in case additional comments are added or concerns
brought up. At that point, I will ask for "final approval" before doing the
merge.

Regards,

John Ganci

**Review by abekornelis — DISMISSED** [review]
Looks good to me.

**Review by dsh33782 — DISMISSED** [review]
Updates ok to me.

**Comment by jyganci** [comment]
Abe and Anthony are getting errors running the DFHALL.BAT file for the
COBOL builds on Windows. I did not have any problems running DFHALL.BAT
(and DFHALLV.BAT) on Windows 10 and Java 17 running on a VM within Virtual
Box. At the present time I do not know the cause of the problems.

Until the problems are resolved, I do not want to merge the pull request.

Regards,

John Ganci

**Comment by adelosa** [comment]
Hi @jyganci - I created a test branch and added the DFHALL.BAT to the GitHub Windows build process - I wanted to see if GitHub runs OK like yours or fails like mine and @abekornelis. 

https://github.com/adelosa/z390/runs/6239292866?check_suite_focus=true 

The build does fail with the same error as me and Abe on TESTBEC1.
This is a pure environment that is recreated for each build.. So my conclusion is something you are doing is making this work.
Maybe we can work on this together - show me what you do and we can try to work out the magic bit.
```
09:38:56 TESTBEC1  MZ390 START USING z390 v1.8.0-46-g273eea7  ON J2SE 1.8.0_322 04/30/22
AZ390E error 198     (43/23)342            LARL  R1,P0202         ADDRESS SOURCE
AZ390E error 198     (43/24)343            LARL  R15,P0202        ADDRESS TARGET
AZ390E error 198     (43/28)344            XC    0(P0202END-P0202,R15),0(R1) XC IT
AZ390E error 198     (44/24)345            LARL  R1,=X'0202'         ADDRESS SOURCE
AZ390E error 198     (44/28)346            LARL  R15,P0202FN        ADDRESS TARGET AS LABEL
AZ390E error 198     (44/31)347            MVC   0(L'P0202FN,R15),0(R1) MVC IT
AZ390E error 198     (42/70)348            LARL  R15,P0202CAP       ADDRESS TARGET
AZ390E error 198     (42/71)349            MVC   0(4,R15),DFHEICAP  SET COMMAREA ADDRESS
AZ390E error 198    (42/106)350            LA    R0,ADDRESS,OF,DFHCWA            ADDRESS/REG OF CWA POINTER
AZ390E error 198     (45/23)352            LARL  R1,P0202ACW         ADDRESS TARGET
AZ390E error 198     (45/24)353            ST    R0,0(R1)      ST IT
AZ390E error 198    (42/129)354            LA    R0,DFHEIHCN        ADDRESS OF DFHEIHCN
AZ390E error 198     (45/23)355            LARL  R1,P0202HCN         ADDRESS TARGET
AZ390E error 198     (45/24)356            ST    R0,0(R1)      ST IT
AZ390E error 198     (45/23)357            LARL  R1,P0202EIB         ADDRESS TARGET
AZ390E error 198     (45/24)358            ST    DFHEIBR,0(R1)      ST IT
AZ390E error 198     (45/23)359            LARL  R1,P0202TCT         ADDRESS TARGET
AZ390E error 198     (45/24)360            ST    TCTTEAR,0(R1)      ST IT
AZ390E error 198     (45/23)361            LARL  R1,P0202EDS         ADDRESS TARGET
AZ390E error 198     (45/24)362            ST    R13,0(R1)      ST IT
AZ390E error 198    (42/135)363            L     R0,DFHEILEN        DSA LENGTH
AZ390E error 198     (46/23)364            LARL  R1,P0202EDL         ADDRESS TARGET
AZ390E error 198     (46/24)365            STH   R0,0(R1)      STH IT
AZ390E error 198    (42/137)366            LA    R0,DFHEIUSR        DSA USER AREA
AZ390E error 198    (42/138)367            SR    R0,R13             R0=CICS DSA LENGTH
AZ390E error 198     (46/23)368            LARL  R1,P0202EDC         ADDRESS TARGET
AZ390E error 198     (46/24)369            STH   R0,0(R1)      STH IT
AZ390E error 198    (42/151)370            LARL  R0,P0202EDF_0250 ADDRESS ORIGIN FOR CEDF
AZ390E error 198     (45/23)371            LARL  R1,P0202EDF         ADDRESS TARGET
AZ390E error 198     (45/24)372            ST    R0,0(R1)      ST IT
AZ390E error 198    (42/153)373            L     R15,TCTTELCL       R15=LCL MODULE INDEXER
AZ390E error 198    (42/154)374            LARL  R1,=A(P0202)       R1=LCL MODULE PARAMETER LIST
AZ390E error 198    (42/156)376            BAKR  0,R15              STACK REGS AND GO
AZ390E error 198    (42/163)377            L     ADDRESS,OF,DFHCWA,TCTTECWA      LOAD CWA ADDRESS
AZ390E error 198     (43/23)409            LARL  R1,P0404         ADDRESS SOURCE
AZ390E error 198     (43/24)410            LARL  R15,P0404        ADDRESS TARGET
AZ390E error 198     (43/28)411            XC    0(P0404END-P0404,R15),0(R1) XC IT
AZ390E error 198     (44/24)412            LARL  R1,=X'0404'         ADDRESS SOURCE
AZ390E error 198     (44/28)413            LARL  R15,P0404FN        ADDRESS TARGET AS LABEL
AZ390E error 198     (44/31)414            MVC   0(L'P0404FN,R15),0(R1) MVC IT
AZ390E error 198     (44/24)415            LARL  R1,=AL2(49)         ADDRESS SOURCE
AZ390E error 198     (44/28)416            LARL  R15,P0404LEN        ADDRESS TARGET AS LABEL
AZ390E error 198     (44/31)417            MVC   0(L'P0404LEN,R15),0(R1) MVC IT
AZ390E error 198    (49/183)418            LA    R0,SENDWCC           FROM ADDRESS
AZ390E error 198     (45/23)419            LARL  R1,P0404FRM         ADDRESS TARGET
AZ390E error 198     (45/24)420            ST    R0,0(R1)      ST IT
AZ390E error 198     (45/23)421            LARL  R1,P0404EIB         ADDRESS TARGET
AZ390E error 198     (45/24)422            ST    DFHEIBR,0(R1)      ST IT
AZ390E error 198     (45/23)423            LARL  R1,P0404TCT         ADDRESS TARGET
AZ390E error 198     (45/24)424            ST    TCTTEAR,0(R1)      ST IT
AZ390E error 198     (45/23)425            LARL  R1,P0404EDS         ADDRESS TARGET
AZ390E error 198     (45/24)426            ST    R13,0(R1)      ST IT
AZ390E error 198    (49/195)427            L     R0,DFHEILEN        DSA LENGTH
AZ390E error 198     (46/23)428            LARL  R1,P0404EDL         ADDRESS TARGET
AZ390E error 198     (46/24)429            STH   R0,0(R1)      STH IT
AZ390E error 198    (49/197)430            LA    R0,DFHEIUSR        DSA USER AREA
AZ390E error 198    (49/198)431            SR    R0,R13             R0=CICS DSA LENGTH
AZ390E error 198     (46/23)432            LARL  R1,P0404EDC         ADDRESS TARGET
AZ390E error 198     (46/24)433            STH   R0,0(R1)      STH IT
AZ390E error 198    (49/212)434            LARL  R0,P0404EDF_0279 ADDRESS ORIGIN FOR CEDF
AZ390E error 198     (45/23)435            LARL  R1,P0404EDF         ADDRESS TARGET
AZ390E error 198     (45/24)436            ST    R0,0(R1)      ST IT
AZ390E error 198    (49/214)437            L     R15,TCTTELCL       R15=LCL MODULE INDEXER
AZ390E error 198    (49/215)438            LARL  R1,=A(P0404)       R1=LCL MODULE PARAMETER LIST
AZ390E error 198    (49/217)440            BAKR  0,R15              STACK REGS AND GO
AZ390E error 198     (50/23)441            LARL  R1,P0404NOH         ADDRESS TARGET
AZ390E error 198     (50/24)442            CLI   0(R1),X'FF'      CLI IT
AZ390E error 198     (43/23)474            LARL  R1,P0E08         ADDRESS SOURCE
AZ390E error 198     (43/24)475            LARL  R15,P0E08        ADDRESS TARGET
AZ390E error 198     (43/28)476            XC    0(P0E08END-P0E08,R15),0(R1) XC IT
AZ390E error 198     (44/24)477            LARL  R1,=X'0E08'         ADDRESS SOURCE
AZ390E error 198     (44/28)478            LARL  R15,P0E08FN        ADDRESS TARGET AS LABEL
AZ390E error 198     (44/31)479            MVC   0(L'P0E08FN,R15),0(R1) MVC IT
AZ390E error 198     (44/24)480            LARL  R1,=CL4'BEC1'         ADDRESS SOURCE
AZ390E error 198     (44/28)481            LARL  R15,P0E08TRN        ADDRESS TARGET AS LABEL
AZ390E error 198     (44/31)482            MVC   0(L'P0E08TRN,R15),0(R1) MVC IT
AZ390E error 198    (52/124)483            LA    R0,MYCOMM       R0=COMMAREA ADDRESS FROM LABEL
AZ390E error 198     (45/23)484            LARL  R1,P0E08COM         ADDRESS TARGET
AZ390E error 198     (45/24)485            ST    R0,0(R1)      ST IT
AZ390E error 198    (52/155)486            LAY   R0,L'MYCOMM     R0=IMPLIED LENGTH OF COMMAREA
AZ390E error 198     (46/23)487            LARL  R1,P0E08CLN         ADDRESS TARGET
AZ390E error 198     (46/24)488            STH   R0,0(R1)      STH IT
AZ390E error 198     (45/23)489            LARL  R1,P0E08EDS         ADDRESS TARGET
AZ390E error 198     (45/24)490            ST    R13,0(R1)      ST IT
AZ390E error 198    (52/200)491            L     R0,DFHEILEN        DSA LENGTH
AZ390E error 198     (46/23)492            LARL  R1,P0E08EDL         ADDRESS TARGET
AZ390E error 198     (46/24)493            STH   R0,0(R1)      STH IT
AZ390E error 198    (52/202)494            LA    R0,DFHEIUSR        DSA USER AREA
AZ390E error 198    (52/203)495            SR    R0,R13             R0=CICS DSA LENGTH
AZ390E error 198     (46/23)496            LARL  R1,P0E08EDC         ADDRESS TARGET
AZ390E error 198     (46/24)497            STH   R0,0(R1)      STH IT
AZ390E error 198    (52/217)499            LARL  R0,P0E08EDF_0293 ADDRESS ORIGIN FOR CEDF
AZ390E error 198     (45/23)500            LARL  R1,P0E08EDF         ADDRESS TARGET
AZ390E error 198     (45/24)501            ST    R0,0(R1)      ST IT
AZ390E error 198    (52/219)502            LARL  R1,P0E08           R1=PARAMETER LIST
AZ390E error 198    (52/220)503            LARL  R14,P0E08RTN_0293 RETURN ADDRESS
AZ390E error 198    (52/221)504            LRL   R15,=V(LCL0E08)    R15=EXTERNAL RETURN MODULE ADDRESS
AZ390E error 198    (52/223)506            BR    R15                GO TO IT
AZ390E error 198     (43/23)584            LARL  R1,P0404         ADDRESS SOURCE
AZ390E error 198     (43/24)585            LARL  R15,P0404        ADDRESS TARGET
AZ390E error 198     (43/28)586            XC    0(P0404END-P0404,R15),0(R1) XC IT
AZ390E abort 49 on line 586          XC    0(P0404END-P0404,R15),0(R1) XC IT
AZ390I max errors exceeded
AZ390E error 198     (44/24)587            LARL  R1,=X'0404'         ADDRESS SOURCE
AZ390I aborting due to recursive abort error 49 - max errors exceeded
09:38:57 TESTBEC1  MZ390 ENDED   RC=16 SEC= 1 MEM(MB)=129 IO=27273
D:\a\z390\z390\bat\MZ390 ERROR: See errors on mz390 generated bal file and console
.\BUILD.BAT ERROR: Encountered RC 16 - exiting
Error: Process completed with exit code 16.

```

**Comment by jyganci** [comment]
As I've said before, I do not build directly using my clone. Here's what I
do. All done from terminal.

1. Zip my clone.
1.1. cd git-workspace
1.2. cd z390 (my z390 clone)
1.3. git status (to verify that the branch is correct; is356 for the
current pull request)
1.4. cd ..
1.5. zip -r z390-git-yyyy-mm-dd.zip z390
2. Move to a work directory. In my case, it is ~/z390/work. My current z390
work directory is here.
2.1. Current z390 directory -- subdirectory of work -- is slightly modified.
2.1.1. z390.properties file resides there. Obtained by extracting it from
original z390.jar file.
2.1.2. Commented line(s) in bash/bldjar and bat/BLDJAR.BAT that calls
relver to build z390.properties.
2.1.3. Commented line(s) in build.sh (none in BUILD.BAT) that build the
distro.
3. Save my current z390 work directory via mv z390 z390-prev-yyyy-mm-dd
4. Copy the zip file here then unzip it; result is a new z390 work
directory.
5. Slightly modify new z390 directory.
5.1. Copy z390.properties file from previous z390 work (saved in step 3)
here.
5.2. Save original build.sh, BUILD.BAT as build.sh.orig, BUILD.BAT.orig
then copy 2.1.2 saved versions here.
5.3. Same thing for bash/bldjar, bat/BLDJAR.BAT.
6. Run build.sh from the new z390 work directory. It succeeds.
7. Run cics/bash/dfhall from the new z390 work directory. It succeeds.
8. Run cics/bash/dfhallv from the new z390 work directory. It succeeds.
9. Run cics/bash/z390cicg to start zCICS. It starts successfully.
10. Test some CICS transactions. All okay. Shut down CICS.

Now that CICS has been verified on Linux, move on to Windows. Note: using
Java 17 with Windows.

11. Copy the zip file to a USB stick. Remaining steps on my MacBook Pro.
12. Copy the zip file to winshare directory. (Directory shared with Windows
10 running in Virtual Box).
13. Save existing z390  directory there with new name. (Contains modified
cics\bat\DFHALL.BAT.)
14. Unzip the file. New z390 directory. Note: this is done while on MacOS.
15. Start Virtual Box. Start Windows 10. Log in. Note: Z: is the winshare
directory.
16. Open command prompt. Change directory to z:\z390.
17. Before running BUILD.BAT, must do one change.
17.1. Open Notepad++. Edit z:\z390\demo\TESTDCB1.TF1.
17.2. Save it with Windows line endings. (Has Linux line endings because
unzip done on MacOS, I think.)
17.3. Had to do this because one of the demo programs failed.
18. At this point, we have a z390 directory with two modifications.
18.1. z390.properties file already exists.
18.2 bat\BLDJAR.BAT has been modified to not call RELVER.BAT.
18.3 Also added something like "-nowarn" to the "javac" line to get around
warning messages about security.
19. Run BUILD.BAT. For me, this was very painful. Many timeouts.
19.1. Note that, logically, I ran BUILD.BAT. Latest run picked up after
first timeout. Ran just that step manually.
19.2. Ran remaining BUILD.BAT steps manually, one at a time.
20. Run cics\bat\DFHALL.BAT.
20.1. Because of MANY timeouts, I manually edited DFHALL.BAT and commented
all the UNREF steps.
20.2. Even then, it timed out near the end (last 3-4 steps). NO PROBLEMS
WITH COBOL STEPS.
20.3. Created DFHALL1.BAT with just the step that timed out, plus the
remaining steips; ran that.
21. Run cics\bat\DFHALLV.BAT. No problems here.
22. Run cics\bat\Z390CICG.BAT to start CICS.
23. Test some transactions. All okay. Shut down CICS.

You may see something wrong here, but I don't. Logically, all I've done is
modify the build process to run without a git dependency by not calling
relver and providing a z390.properties file.

Regards,

John Ganci

**Review by adelosa — APPROVED** [review]
Happy to approve this.

**Comment by jyganci** [comment]
Abe, Don,All

Thanks for the comments.

New/updated copyrights have been pushed to my fork for

DFHPCT.MLC
TESTGUI7.MLC
TESTGUIC.CBL
TESTNT01.MLC
TESTNT02.MLC

I agree with Don's comments about UNREF. I would prefer remove/update it
via a separate issue. Comments?

Regards,

John Ganci

**Pull request merged**